### PR TITLE
types: Minor fixes to sparse function indices and implicit dims

### PR DESCRIPTION
**Pull request by mloubout**
- Fixes remaining hardcoded `-1` for the position of the sparse index
- - Fix missing implicit dims for the final equation of interpolation

Added test for it

### CODE REVIEW

File: devito/operations/interpolators.py [hunk 1]
```diff
@@ -256,7 +256,8 @@ def callback():
 
             # Write/Incr `self`
             lhs = self.sfunction.subs(self_subs)
-            last = [Inc(lhs, rhs)] if increment else [Eq(lhs, rhs)]
+            last = [Inc(lhs, rhs, implicit_dims=implicit_dims)] if increment else \
```
Comment (FabioLuporini) [inline devito/operations/interpolators.py:259]: imho more readable as:

```
if increment:
    cls = Inc
else:
    cls = eq
last = [cls(lhs, rhs, ...)]
```

File: tests/test_interpolation.py [hunk 2]
```diff
@@ -14,6 +14,11 @@
 import scipy.sparse
 
 
+class SparseFirst(SparseFunction):
```
Comment (FabioLuporini) [inline tests/test_interpolation.py:17]: why not making this  test-local ?

**Review by FabioLuporini — APPROVED** [review]
great

File: devito/operations/interpolators.py (hunk 1 above)
Comment (mloubout) [reply devito/operations/interpolators.py:259]: Yeah thought about doing it and forgot before pushing will fix

File: tests/test_interpolation.py (hunk 2 above)
Comment (mloubout) [reply tests/test_interpolation.py:17]: Wasn't sure if was gonna need more tests for it in the future (I'm sure someone will come up with more bugs) so thought was better but happy to move it for now

File: tests/test_interpolation.py [hunk 3]
```diff
@@ -590,3 +590,38 @@ def test_msf_interpolate():
...
+
+    # Operator
+    u = TimeFunction(name="u", grid=grid, time_order=1)
+    fs = Function(name="fs", grid=grid, dimensions=(*dims, ds), shape=(11, 11, 3))
+
+    eqs = [Eq(u.forward, u+1), Eq(fs, u)]
+    # No time dependence so need the implicit dim
+    rec = s.interpolate(expr=s+fs, implicit_dims=grid.stepping_dim)
+    op = Operator(eqs + rec)
+
```
Comment (georgebisbas) [inline tests/test_interpolation.py:624]: I would add a check of the order of dimensions:
using assert_structure(op, ....)

**Review by georgebisbas — COMMENTED** [review]
Good thanks, left a comment to add a test

File: tests/test_interpolation.py (hunk 3 above)
Comment (mloubout) [reply tests/test_interpolation.py:624]: There is still some simd warning and op "issues" to be resolved, this mainly makes sure result is correct and that the legacy hardcoded value are fixed.

**Pull request merged**